packages/@react-spectrum/list/src/styles.css

```diff
@@ -343,6 +346,12 @@
     font-size: var(--spectrum-listview-item-title-text-size);
   }
 
+  &:not(.react-spectrum-ListView--hasDescription) {
+    .react-spectrum-ListViewItem-content {
+      grid-area: content / description;
```

Comment:
for anyone else reviewing, we had to do it this way because the grid allocated space in the second row
see https://jsfiddle.net/t5c2hjz0/2/ to play with it, try changing grid-template-rows to `  grid-template-rows: auto minmax(0, auto);`

I'm unable to determine where the extra white space is coming from in either case.